File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
       arr = []
       cur = head
       while cur:
        arr.append(cur)
        cur = cur.next
       cur = arr[0]
       prev = cur
       dummy = ListNode()
       dummy.next = cur

       if len(arr) - n == 0:
        dummy.next = cur.next
        return dummy.next
    
       
       for i in range(len(arr)+1):
        if i == len(arr)-n:
            print("index", i, "v", arr[i].val)
            print(prev.val)
            print(cur.val)  
            cur.next = arr[i].next
            break
        else:
            cur.next = arr[i]
            prev = cur
            cur = cur.next
       return dummy.next
    
        

        

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode(-1);
        ListNode list3 = dummy;
        while(list1 != null && list2 != null) {
            
            if(list1.val > list2.val) {
                dummy.next = list2;
                dummy = dummy.next;
                list2 = list2.next;

            } else if(list1.val < list2.val) {
                dummy.next = list1;
                dummy = dummy.next;
                list1 = list1.next;
            } else {
                dummy.next = list1;
                dummy = dummy.next;
                list1 = list1.next;
            }
        }
        if(list1 != null) {
            while(list1 != null) {
                dummy.next = list1;
                dummy = dummy.next;
                list1 = list1.next;
            }
            
        }

        if(list2 != null) {
            while(list2 != null) {
                dummy.next = list2;
                dummy = dummy.next;
                list2 = list2.next;
            }
        }



        return list3.next;

    }
}